### PR TITLE
Appointment list

### PR DESCRIPTION
**Pull request by LomyW**
todo: some mini stylings and - maybe - renaming 😄

### CODE REVIEW

File: src/NavigatorLazy.tsx [hunk 1]
```diff
@@ -250,6 +251,15 @@ export default function NavigatorLazy() {
                 }
             ></Route>
 
+            <Route
+                path="/appointments"
+                element={
+                    <RequireAuth>
+                        <SwitchUserType pupilComponent={<Appointments />} studentComponent={<Appointments />} />
```
Comment (jeffersondarcy) [inline src/NavigatorLazy.tsx:258]: do we need this switch?

File: src/helper/appointment-helper.ts [hunk 2]
```diff
@@ -53,4 +55,36 @@ const getCourseDay = (courseDate: string): CourseDay => {
     return { courseDay, courseDateDay };
 };
 
-export { getCourseDay, isCourseTakingPlaceRightNow, getCourseTimeText };
+const getNextCourseId = (appointments: AppointmentType[]): number => {
+    const now = DateTime.now();
+    const nextCourse = appointments.find((appointment) => {
+        console.log(DateTime.fromISO(appointment.startDate) > now);
```
Comment (jeffersondarcy) [inline src/helper/appointment-helper.ts:61]: log should be removed

File: src/helper/appointment-helper.ts [hunk 3]
```diff
@@ -53,4 +55,36 @@ const getCourseDay = (courseDate: string): CourseDay => {
...
+    if (sameMonth && sameYear) return true;
+    return false;
+};
+
+const getScrollToId = (): number => {
+    const currentId = appointments.currentCourseId;
+    const nextId = appointments.nextCourseId;
+
+    if (currentId) {
+        console.log('current ' + currentId);
```
Comment (jeffersondarcy) [inline src/helper/appointment-helper.ts:81]: do you want to keep this logs?

File: src/widgets/appointment/Appointment.tsx [hunk 4]
```diff
@@ -1,42 +1,62 @@
...
+    firstname: string;
+};
+
+const Appointment: React.FC<Props> = ({ courseStart, duration, courseTitle, instructors, participants, scrollToRef }) => {
     const [isCurrent, setIsCurrent] = useState<boolean>(false);
+    const currentMonth = isCurrentMonth(courseStart);
 
+    // TODO we have to update the effect more often so that if a course will get "current" the UI changes
     useEffect(() => {
         setIsCurrent(isCourseTakingPlaceRightNow(courseStart, duration));
```
Comment (jeffersondarcy) [inline src/widgets/appointment/Appointment.tsx:32]: I am not sure about this logic. Does isCurrent need to be in state - it is only changing with changing props and not by a user-interaction

File: src/widgets/appointment/Appointment.tsx [hunk 5]
```diff
@@ -1,42 +1,62 @@
...
+type Participant = {
+    firstname: string;
+};
+
+const Appointment: React.FC<Props> = ({ courseStart, duration, courseTitle, instructors, participants, scrollToRef }) => {
     const [isCurrent, setIsCurrent] = useState<boolean>(false);
+    const currentMonth = isCurrentMonth(courseStart);
 
+    // TODO we have to update the effect more often so that if a course will get "current" the UI changes
     useEffect(() => {
```
Comment (jeffersondarcy) [inline src/widgets/appointment/Appointment.tsx:31]: I don't think useEffect is needed here

File: src/widgets/appointment/Appointment.tsx [hunk 6]
```diff
@@ -1,42 +1,62 @@
...
+    firstname: string;
+    lastname: string;
+};
+
+type Participant = {
+    firstname: string;
+};
+
+const Appointment: React.FC<Props> = ({ courseStart, duration, courseTitle, instructors, participants, scrollToRef }) => {
     const [isCurrent, setIsCurrent] = useState<boolean>(false);
```
Comment (jeffersondarcy) [inline src/widgets/appointment/Appointment.tsx:27]:
```suggestion
    const isCurrent = isCourseTakingPlaceRightNow(courseStart, duration));
```

Comment (jeffersondarcy) [reply src/widgets/appointment/Appointment.tsx:27]: can we do it without state?

**Review by jeffersondarcy — CHANGES_REQUESTED** [review]
please check if state and useEffect are required in Appointment.tsx

**Review by timidak — APPROVED** [review]
👍🏼 LGTM

**Pull request merged**